File: Python/String-2/cat_dog.py
def cat_dog(str):
  c_count = 0
  d_count = 0
  
  for i in range(len(str) - 2):
    if (str[i] + str[i + 1] + str[i + 2] == "cat"):
      c_count += 1
    if (str[i] + str[i + 1] + str[i + 2] == "dog"):
      d_count += 1
  return c_count == d_count

File: Java/String-2/catDog.java
public boolean catDog(String str) {
  int c_count = 0, d_count = 0;
  for (int i = 0; i < str.length() - 2; i++) {
    if (str.substring(i, i + 3).equals("cat")) c_count++;
    if (str.substring(i, i + 3).equals("dog")) d_count++;
  }
  return c_count == d_count;
}
